### PR TITLE
🌱 chore: use capo cluster agent for API requests

### LINKED ISSUE #1611: Use reasonable User-Agent for requests
/kind bug

**What steps did you take and what happened:**
No details for the version of CAPO in the User Agent.

**What did you expect to happen:**
It would be very nice to be able to have the CAPO version in the User Agent to simplify troubleshooting.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built):
- Cluster-API version:
- OpenStack version:
- Minikube/KIND version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by mnaser**
**What this PR does / why we need it**:
At the moment, we do not modify the user agent for all API requests we make which make it hard to identify if certain requests were done by CAPO or not.

This change prepends the CAPO version to the Gophercloud user agent which will give plenty of benefits.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1611 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

**Comment by jichenjc** [comment]
/lgtm

make paste the updated output will be informative

**Review by mdbooth — COMMENTED** [review]
This looks great. Unfortunately we don't currently set the version in our builds, so we're going to have to fix that too.

For an example of how this is done see: https://github.com/kubernetes/cloud-provider-openstack/blob/b229f849774e7e1976256faf2df4931841d242fa/Makefile#L39

**Comment by mnaser** [comment]
@mdbooth ... i think we do..?

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/Makefile#L114-L115

though i haven't really validated here.

**Comment by mdbooth** [comment]
> @mdbooth ... i think we do..?
> 
> https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/Makefile#L114-L115
> 
> though i haven't really validated here.

I didn't know that was there!

/approve

File: pkg/scope/provider.go [hunk 1]
```diff
@@ -152,6 +153,10 @@ func NewProviderClient(cloud clientconfig.Cloud, caCert []byte, logger logr.Logg
 		return nil, nil, "", fmt.Errorf("create providerClient err: %v", err)
 	}
 
+	ua := gophercloud.UserAgent{}
+	ua.Prepend(fmt.Sprintf("cluster-api-provider-openstack/%s", version.Get().String()))
```
Comment (stephenfin) [inline pkg/scope/provider.go:157]: Is there benefit in using the default user agent and simply prepending to it? That way we'll also get the gophercloud version in use. Perhaps that's overkill...?

Comment (mnaser) [reply pkg/scope/provider.go:157]: Personally I mimicked the patter that was used by the CPO so that's how I got where I got.

**Pull request merged**